qiskit_experiments/curve_analysis/curve_analysis.py

```diff
@@ -178,64 +184,44 @@ def _run_data_processing(
                 ("xval", float),
                 ("yval", float),
                 ("yerr", float),
-                ("model_name", "U30"),  # builtin str is U0 which is zero-length unicode in numpy
-                ("model_id", int),
+                ("name", "U30"),
+                ("class_id", int),
+                ("category", "U30"),
                 ("shots", int),
             ]
         )
-        table_data = np.empty(len(to_process), dtype=dtypes)
+
+        # Prepare circuit metadata to data class mapper from data_subfit_map value.
+        classifier = {}
+        if len(self._models) == 1:
+            classifier[self.model_names()[0]] = {}
```

Comment:
Minor clean up since `classifier` is only set once it doesn't need to start as `{}`.

```suggestion
        if len(self._models) == 1:
            classifier = {self.model_names()[0]: {}}
```